src/util/string/hash.js

```diff
@@ -0,0 +1,12 @@
+define(function() {
+
+// Based on http://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript-jquery
+return function( str ) {
+	return [].reduce.call( str, function( hash, i ) {
```

Comment:
As far as I can tell, this will get called pretty often, so its probably worth optimizing. You can avoid lots of function calls why using a regular loop, ala https://github.com/jquery/qunit/blob/29ca2e87c420f6dfdef793b034a67cff9a666df5/src/test.js#L432-L435